grades/api.py

```diff
@@ -177,11 +177,14 @@ def freeze_user_final_grade(user, course_run, raise_on_exception=False):
                     course_run.edx_course_key
                 )
             ) from ex
-    return FinalGrade.objects.create(
+    # the final grade at this point should not exists, but putting a `get_or_create`
+    # should solve the problem when the function is called synchronously from the dashboard REST API multiple times
+    final_grade_obj, _ = FinalGrade.objects.get_or_create(
```

Comment:
I think that if 2 (assumed concurrent) requests to freeze the grade give different results, we have a much bigger problem and I probably would like to see it.